src/dream.mli

```diff
@@ -2193,6 +2193,7 @@ val catch : (error -> response promise) -> middleware
 val run :
   ?interface:string ->
   ?port:int ->
+  ?socket_path:string ->
```

Comment:
I wonder if it would be a cleaner API to offer separate functions for serving on a port (which would likely be used more) vs on a "network address", which could correspond to the `network` type defined elsewhere in this PR? Feels like we shouldn't accept multiple config parameters if some of them could get cancelled out. Possibly `run_sock` and `serve_sock`?

Doesn't look like `interface` is needed if listening on a unix socket either; pretty sure `is_localhost` would just need to be adapted to return `true` if on a socket.

Reply:
Maybe a `socket_source` instead of `interface`, `port` or `socket_path` can be passed. `Inet` or `Unix` being possible kinds of sources. Although that will make the run function slightly more cumbersome, but the implementation can just default to `Inet(defaultInterface, defaultPort)`.